pkg/network/tracer/tracer_usm_linux_test.go

```diff
@@ -254,14 +251,30 @@ func testHTTPSLibrary(t *testing.T, fetchCmd []string, prefetchLibs []string) {
 			statsTags := req.StaticTags
 			// debian 10 have curl binary linked with openssl and gnutls but use only openssl during tls query (there no runtime flag available)
 			// this make harder to map lib and tags, one set of tag should match but not both
-			if key.Path.Content == "/200/foobar" && (statsTags == tagGnuTLS || statsTags == tagOpenSSL) {
-				t.Logf("found tag 0x%x %s", statsTags, staticTags[statsTags])
-				return true
+			foundPathAndHTTPTag := false
+			if key.Path.Content == "/200/foobar" && (statsTags == network.ConnTagGnuTLS || statsTags == network.ConnTagOpenSSL) {
+				foundPathAndHTTPTag = true
+				t.Logf("found tag 0x%x %s", statsTags, network.GetStaticTags(statsTags))
+			}
+
+			if foundPathAndHTTPTag {
+				// socket filter is not supported on fentry tracer
+				if fentryTracerEnabled {
+					// so we return early if the test was successful until now
+					return true
+				}
+
+				for _, c := range payload.Conns {
+					if c.SPort == key.SrcPort && c.DPort == key.DstPort && isTLSTag(c.StaticTags) {
+						return true
+					}
+				}
+				t.Logf("HTTP connection %v doesn't contain ConnTagTLS\n", key)
```

Comment:
consider the following simplification:
```suggestion
			if key.Path.Content == "/200/foobar" && (statsTags == network.ConnTagGnuTLS || statsTags == network.ConnTagOpenSSL) {
				t.Logf("found tag 0x%x %s", statsTags, network.GetStaticTags(statsTags))
				// socket filter is not supported on fentry tracer
				if fentryTracerEnabled {
					// so we return early if the test was successful until now
					return true
				}

				for _, c := range payload.Conns {
					if c.SPort == key.SrcPort && c.DPort == key.DstPort && isTLSTag(c.StaticTags) {
						return true
					}
				}
				t.Logf("HTTP connection %v doesn't contain ConnTagTLS\n", key)
		}
```

Reply:
done